std/typecons.d

```diff
@@ -9035,3 +9035,220 @@ unittest
     assert((a ^ true) == Ternary.no);
     assert((a ^ false) == Ternary.yes);
 }
+
+/**
+Provides untyped storage of size `Size` aligned to `Alignment`.
+
+`Alignment` must be a power of 2.
+Default `Alignment` is the maximum alignment requirement on the current platform.
+
+Note: while this type ensures storage alignment on its own, it still obeys
+language rules. Therefore, care must be taken when using it in aggregate types,
+notably with custom `align` specification.
+
+Since aggregated object cannot test its alignment in an aggregate
+at compile time, an assert will be issued at runtime when a misaligned
+AlignedStorage is used.
+*/
+struct AlignedStorage(size_t Size, size_t Alignment = platformAlignment)
```

Comment:
This somewhat sophisticated API is ill-advised. We're looking at an "abstraction that doesn't abstract": a data member that is nominally private, yet trivially made public by means of accessors. There's no need for APIs for length, slicing etc because all they do is morally equivalent with:

```d
struct AlignedStorage(size_t Size, size_t Alignment = platformAlignment)
{
    align(Alignment) void[Size] payload;
}
```

Everything needed is readily accessible by means of `.payload`. This also emphasizes the question of why we need to put in the standard library what's essentially a three-line trick.

Reply:
...with the exception of an invariant that safeguards against misaligned moves :)